### PR TITLE
fixing action

### LINKED ISSUE #456: Dependency Graph is missing S3 scan object Dockerfile
# Summary
For some reason, the S3 scan object Dockerfile dependencies are not part of the repo's Dependency Graph.  They look to have been successfully added by this workflow:
https://github.com/cds-snc/scan-files/actions/runs/3565853924/jobs/5991547797

```sh
trivy image \
    --format github \
    --vuln-type os,library \
    --security-checks vuln \
    --output dependency-results.sbom.json \
    ***.dkr.ecr.ca-central-1.amazonaws.com/scan-files/module/s3-scan-object:8af10bb2068481cdcbcff579ad1de796ada0b6c5

curl \
    -H 'Accept: application/vnd.github+json' \
    -H 'Authorization: token ***' \
    'https://api.github.com/repos/'$GITHUB_REPOSITORY'/dependency-graph/snapshots' \
    -d @dependency-results.sbom.json

{
  "id": 429611,
  "created_at": "2022-11-28T14:50:30.409Z",
  "result": "SUCCESS",
  "message": "Dependency results for the repo have been successfully updated."
}
```

However, as of now, they're not part of the graph.

# Related
- cds-snc/platform-core-services#181

### PR DESCRIPTION
**Pull request by ascheid**
# Summary | Résumé

- Creation of a variable to define if we would like to upload the vulnerabilities scan result to GitHub Security. When we rollout, the idea is that we would start not uploading the results to not overwhelm teams. If nothing is passed to this input, upload the results is going to happen by default.
- Retag the docker image to increase readability and easier relate to the image being scanned.
- Changes with sbom upload. We only want to submit the software bill of material when there is a push to the default branch.

### CODE REVIEW

**Comment by ascheid** [comment]
- Closes https://github.com/cds-snc/scan-files/issues/456

File: .github/actions/docker-scan/action.yml [hunk 1]
```diff
@@ -11,42 +11,56 @@ inputs:
...
+        docker tag ${{ inputs.docker_image }} ${{ inputs.sbom_name }}
+      shell: bash
+
     - name: Install Trivy
       run: |
         curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sh -s -- -b /usr/local/bin v0.34.0
       shell: bash
 
     - name: run trivy docker
+      if: github.event_name == 'push' && format('refs/heads/{0}', github.event.repository.default_branch) == github.ref
```
Comment (patheard) [inline .github/actions/docker-scan/action.yml]: Could you switch up this to capture this condition in an environment variable and then check that var in all the subsequent `if` conditions:

```yaml
  - name: set upload sbom conditional
    run: |
      echo "upload_sbom=${{ github.event_name == 'push' && format('refs/heads/{0}', github.event.repository.default_branch) == github.ref }}" >> $GITHUB_ENV
```

You should then be able to check it like so on the next steps:

```yaml
    - name: run trivy docker
      if: env.upload_sbom == 'true'
```

Comment (ascheid) [reply .github/actions/docker-scan/action.yml]: done!

File: .github/actions/docker-scan/action.yml [hunk 2]
```diff
@@ -57,13 +79,14 @@ runs:
...
+      if: inputs.upload_docker_vulns == 'true'
       with:
-        image-ref: '${{ inputs.docker_image }}'
+        image-ref: '${{ inputs.sbom_name }}'
         format: 'sarif'
         output: 'trivy-results.sarif'
 
     - name: upload trivy scan results to github security tab
-      uses: github/codeql-action/upload-sarif@312e093a1892bd801f026f1090904ee8e460b9b6 # v2.1.34
+      if: inputs.upload_docker_vulns == 'true'uses: github/codeql-action/upload-sarif@312e093a1892bd801f026f1090904ee8e460b9b6 # v2.1.34
```
Comment (patheard) [inline .github/actions/docker-scan/action.yml]: looks like a line break got deleted accidentally

Comment (ascheid) [reply .github/actions/docker-scan/action.yml]: Done!

**Review by patheard — APPROVED** [review]

**Pull request merged**